<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NewDataTypeForm</class>
 <widget class="QWidget" name="NewDataTypeForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>722</width>
    <height>657</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::TabFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frameForUnlockButton">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnLock">
        <property name="text">
         <string/>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnUndelete">
        <property name="toolTip">
         <string>This object was deleted in a Patch. Click this button to Undelete it.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../dbm.qrc">
          <normaloff>:/images/actions/images/small/gnome_undelete.png</normaloff>:/images/actions/images/small/gnome_undelete.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpContent">
     <property name="title">
      <string/>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLineEdit" name="txtDTName">
        <property name="toolTip">
         <string>The name of the Data Type</string>
        </property>
        <property name="placeholderText">
         <string>Name</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <property name="documentMode">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="tabWidgetPage1">
         <attribute name="icon">
          <iconset resource="../dbm.qrc">
           <normaloff>:/images/actions/images/big/bigdatatypes.png</normaloff>:/images/actions/images/big/bigdatatypes.png</iconset>
         </attribute>
         <attribute name="title">
          <string>Properties</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Type</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cmbDTType">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <item>
             <property name="text">
              <string>Text/String</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_text.png</normaloff>:/images/actions/images/small/datatypes_text.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Numeric</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_numeric.png</normaloff>:/images/actions/images/small/datatypes_numeric.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Boolean</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_bool.png</normaloff>:/images/actions/images/small/datatypes_bool.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Date/Time</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_datetime.png</normaloff>:/images/actions/images/small/datatypes_datetime.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Blob/Binary</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_blob.png</normaloff>:/images/actions/images/small/datatypes_blob.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Misc</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_misc.png</normaloff>:/images/actions/images/small/datatypes_misc.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Spatial</string>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/datatypes_spatial.png</normaloff>:/images/actions/images/small/datatypes_spatial.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>SQL</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="cmbDTSQLType"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="lblWidth">
            <property name="text">
             <string>Width</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="txtWidth"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="lblValues">
            <property name="text">
             <string>Values</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <layout class="QGridLayout" name="layoutEnums">
            <item row="0" column="0">
             <widget class="QListWidget" name="lstEnumValues"/>
            </item>
            <item row="0" column="1">
             <widget class="QToolButton" name="btnRemoveEnumValue">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../dbm.qrc">
                <normaloff>:/images/actions/images/small/remove.png</normaloff>:/images/actions/images/small/remove.png</iconset>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QToolButton" name="btnAddEnumValue">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../dbm.qrc">
                <normaloff>:/images/actions/images/small/add.png</normaloff>:/images/actions/images/small/add.png</iconset>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLineEdit" name="txtEnumCurrentValue"/>
            </item>
           </layout>
          </item>
          <item row="11" column="1">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QComboBox" name="cmbEnumItems"/>
            </item>
            <item>
             <widget class="QLineEdit" name="txtDefaultValue"/>
            </item>
            <item>
             <widget class="QCheckBox" name="chkNullIsDefault">
              <property name="text">
               <string>Null</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="14" column="1">
           <widget class="QCheckBox" name="chkUnsigned">
            <property name="text">
             <string>Unsigned</string>
            </property>
           </widget>
          </item>
          <item row="17" column="1">
           <widget class="QCheckBox" name="chkCanBeNull">
            <property name="text">
             <string>Can be NULL</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="19" column="1">
           <widget class="QTextEdit" name="txtDescription">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="lblValues_2">
            <property name="text">
             <string>Default</string>
            </property>
           </widget>
          </item>
          <item row="19" column="0">
           <widget class="QLabel" name="lblValues_3">
            <property name="text">
             <string>Description</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolButton" name="btnSave">
           <property name="toolTip">
            <string>&lt;b&gt; Save &lt;/b&gt; the datatype</string>
           </property>
           <property name="text">
            <string>Save datatype</string>
           </property>
           <property name="icon">
            <iconset resource="../dbm.qrc">
             <normaloff>:/images/actions/images/actions/filesave.png</normaloff>:/images/actions/images/actions/filesave.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>16</width>
             <height>24</height>
            </size>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="btnDeleteDataType">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Delete &lt;/span&gt;this datatype&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Delete Datatype</string>
           </property>
           <property name="icon">
            <iconset resource="../dbm.qrc">
             <normaloff>:/images/actions/images/actions/remove-icon.png</normaloff>:/images/actions/images/actions/remove-icon.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>16</width>
             <height>24</height>
            </size>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="btnCancelEditing">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Delete &lt;/span&gt;this datatype&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Cancel Editing</string>
           </property>
           <property name="icon">
            <iconset resource="../dbm.qrc">
             <normaloff>:/images/actions/images/small/cancel.png</normaloff>:/images/actions/images/small/cancel.png</iconset>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
     <zorder>tabWidget</zorder>
     <zorder>txtDTName</zorder>
     <zorder>widget</zorder>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../dbm.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>cmbDTType</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>basicDTselected(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbDTSQLType</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onSqlTypeSelected(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnAddEnumValue</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onAddMiscValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnRemoveEnumValue</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onRemoveSelectedMiscValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnLock</sender>
   <signal>toggled(bool)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onLockUnlock(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnUndelete</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onUndelete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnDeleteDataType</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onDeleteDataType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSave</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkNullIsDefault</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onNullClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>551</x>
     <y>456</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkCanBeNull</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onCanBeNullClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>551</x>
     <y>537</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtDTName</sender>
   <signal>textEdited(QString)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onNameEdited(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtWidth</sender>
   <signal>textEdited(QString)</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onWidthChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>551</x>
     <y>209</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkUnsigned</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onChkUnsignedClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>551</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnCancelEditing</sender>
   <signal>clicked()</signal>
   <receiver>NewDataTypeForm</receiver>
   <slot>onReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>414</x>
     <y>630</y>
    </hint>
    <hint type="destinationlabel">
     <x>360</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>basicDTselected(QString)</slot>
  <slot>onSave()</slot>
  <slot>onReset()</slot>
  <slot>onSqlTypeSelected(QString)</slot>
  <slot>onAddMiscValue()</slot>
  <slot>onRemoveSelectedMiscValue()</slot>
  <slot>onCharacterSetCmbChanged(int)</slot>
  <slot>onNullClicked()</slot>
  <slot>onCanBeNullClicked()</slot>
  <slot>onHelp()</slot>
  <slot>onLockUnlock(bool)</slot>
  <slot>onUndelete()</slot>
  <slot>onDeleteDataType()</slot>
  <slot>onNameEdited(QString)</slot>
  <slot>onWidthChanged(QString)</slot>
  <slot>onChkUnsignedClick()</slot>
 </slots>
</ui>
